src/pages/CommentVisit.tsx

```diff
@@ -0,0 +1,128 @@
+import DatePicker from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
+import { useState } from 'react';
+import { usersVisits } from '../utils/data';
+import { Box, Button, Flex, FormLabel, Select } from '@chakra-ui/react';
+import { FlexForLabelSxStyle, labelSxStyle, mainBoxSxStyle } from './RegisterVisit';
+import SingleSelect from '../components/SingleSelect';
+import { UserVisit, Visit } from '../utils/types';
+import SingleSelectForCombinedArrays from '../components/SingleSelectForCombinedArrays';
+import SingleTextArea from '../components/SingleTextArea';
+
+export default function CommentVisit() {
+  const [databaseUsersVisits, setDatabaseUsersVisits] = useState<UserVisit[]>(usersVisits);
+  const [selectedUser, setSelectedUser] = useState('');
+  const [selectedVisitForUser, setSelectedVisitForUser] = useState<Visit[]>();
+  const [visitArray, setVisitArray] = useState<string[]>();
+  const [textForComment, setTextForComment] = useState('');
+
+  const handleChangeSelectedUser = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    if (e.target.value.length === 0) {
+      setSelectedVisitForUser([]);
+      return;
+    } else {
+      setSelectedUser(e.target.value);
+      const newArrayUser = usersVisits.filter((el) => el.name === e.target.value);
+      setSelectedVisitForUser(newArrayUser[0].visits);
+      if (newArrayUser[0].visits.length === 0) {
+        return;
+      } else {
+        setVisitArray([newArrayUser[0].visits[0].doctor, newArrayUser[0].visits[0].id.toString()]);
+      }
+    }
+  };
+
+  const handleChangeSelectedVisitForUser = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    const newArrayVisit = e.target.value.split(' ');
+    if (newArrayVisit.length === 0 || e.target.value.length === 0) return;
+    setVisitArray(newArrayVisit);
+  };
+
+  const handleTextAreaChange = (e: React.ChangeEventHandler<HTMLTextAreaElement>) => {
+    // jak to otypowac?
+    setTextForComment(e.currentTarget.value);
+  };
+  const handleSubmitComment = (e: React.MouseEvent<HTMLElement>) => {
+    e.preventDefault();
+
+    if (textForComment === '' || visitArray === undefined || selectedVisitForUser === undefined)
+      return;
+
+    usersVisits.map((el) => {
+      if (el.name === selectedUser) {
+        return el.visits.map((el) => {
+          if (el.id === Number(visitArray[1]) && el.doctor === visitArray[0]) {
+            el.comment = textForComment;
+            //ten zapis nic nie zmienial, jak to lepiej zapisac?
```

Comment:
sprecyzuj pytanie proszę

Reply:
chcialem to zrobic tak:
 return { ...el, comment: textForComment };
 
 ale i tak nie dziala :/

Reply:
tutaj tak właściwie mapujesz i nic z tymi danymi co otrzymasz z mapa nie robisz więc ten kod się odpala ale nic nie robi